adapter/redisquota/redisquota.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package redisquota provides a quota implementation with redis as backend.
+// The prerequisite is to have a redis server running.
+package redisquota
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	ptypes "github.com/gogo/protobuf/types"
+
+	"istio.io/mixer/adapter/redisquota/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/pool"
+)
+
+type builder struct{ adapter.DefaultBuilder }
+
+type redisQuota struct {
```

Comment:
Need some tests for this code. You can try and reuse some of the memQuota tests. If you can figure out how to make the memQuota tests more generic so they can be reused, that'd be great (but not necessary in this PR, if it's too much hassle)